<p>The sports clothing and athleisure market is substantial, boasting a valuation of around $193 billion in 2021, and it anticipates robust growth in the coming decade!

Within this document, our role is that of a product analyst for an online sports clothing company. The company has a targeted focus on enhancing its revenue streams. Our exploration will delve into various product metrics, including pricing, reviews, descriptions, ratings, as well as revenue and website traffic. The goal is to formulate actionable recommendations for the marketing and sales teams.</p>
<p>The database provided to us, <code>sports</code>, contains five tables, with <code>product_id</code> being the primary key for all of them: </p>
<h3 id="info"><code>info</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_name</code></td>
<td><code>varchar</code></td>
<td>Name of the product</td>
</tr>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>description</code></td>
<td><code>varchar</code></td>
<td>Description of the product</td>
</tr>
</tbody>
</table>
<h3 id="finance"><code>finance</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>listing_price</code></td>
<td><code>float</code></td>
<td>Listing price for product</td>
</tr>
<tr>
<td><code>sale_price</code></td>
<td><code>float</code></td>
<td>Price of the product when on sale</td>
</tr>
<tr>
<td><code>discount</code></td>
<td><code>float</code></td>
<td>Discount, as a decimal, applied to the sale price</td>
</tr>
<tr>
<td><code>revenue</code></td>
<td><code>float</code></td>
<td>Amount of revenue generated by each product, in US dollars</td>
</tr>
</tbody>
</table>
<h3 id="reviews"><code>reviews</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_name</code></td>
<td><code>varchar</code></td>
<td>Name of the product</td>
</tr>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>rating</code></td>
<td><code>float</code></td>
<td>Product rating, scored from <code>1.0</code> to <code>5.0</code></td>
</tr>
<tr>
<td><code>reviews</code></td>
<td><code>float</code></td>
<td>Number of reviews for the product</td>
</tr>
</tbody>
</table>
<h3 id="traffic"><code>traffic</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>last_visited</code></td>
<td><code>timestamp</code></td>
<td>Date and time the product was last viewed on the website</td>
</tr>
</tbody>
</table>
<h3 id="brands"><code>brands</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>brand</code></td>
<td><code>varchar</code></td>
<td>Brand of the product</td>
</tr>
</tbody>
</table>

## 1. Counting missing values


<p>Our analysis will involve addressing missing data and working with numeric, string, and timestamp data types to extract insights about the products in the online store. To begin, let's assess the completeness of the data.</p>

In [ ]:
%%sql
postgresql:///sports

-- Count all columns as total_rows
-- Count the number of non-missing entries for description, listing_price, and last_visited
-- Join info, finance, and traffic

SELECT
    COUNT(*) AS total_rows,
    SUM(CASE WHEN i.description IS NOT NULL THEN 1 ELSE 0 END) AS count_description,
    SUM(CASE WHEN f.listing_price IS NOT NULL THEN 1 ELSE 0 END) AS count_listing_price,
    SUM(CASE WHEN t.last_visited IS NOT NULL THEN 1 ELSE 0 END) AS count_last_visited
FROM info i
JOIN finance f
    ON i.product_id = f.product_id
JOIN traffic t
    ON i.product_id = t.product_id;


1 rows affected.


[(3179, 3117, 3120, 2928)]

## 2. Pricing Comparison: Nike vs. Adidas
<p>The total product count in the database is 3,179. Among the columns we have previewed, only one — <code>last_visited</code> — has more than five percent of its values missing. Now, let's shift our focus to pricing.</p>
<p>How do the price points of Nike and Adidas products differ? Addressing this question allows us to construct a comprehensive view of the company's product range and customer market. We will execute a query to generate a distribution of the <code>listing_price</code> and the corresponding count for each price, grouped by <code>brand</code>.</p>

In [ ]:
%%sql

-- Select the brand, listing_price as an integer, and a count of all products in finance
-- Join brands to finance on product_id
-- Filter for products with a listing_price greater than zero
-- Aggregate results by brand and listing_price, and sort the results by listing_price in descending order

SELECT
    b.brand,
    CAST(f.listing_price AS INTEGER) AS listing_price,
    COUNT(f.product_id) AS product_count
FROM brands b
JOIN finance f
    ON b.product_id = f.product_id
WHERE f.listing_price > 0
GROUP BY b.brand, listing_price
ORDER BY listing_price DESC;


 * postgresql:///sports
77 rows affected.


[('Adidas', 300, 2),
 ('Adidas', 280, 4),
 ('Adidas', 240, 5),
 ('Adidas', 230, 8),
 ('Adidas', 220, 11),
 ('Adidas', 200, 8),
 ('Nike', 200, 1),
 ('Adidas', 190, 7),
 ('Nike', 190, 2),
 ('Adidas', 180, 34),
 ('Nike', 180, 4),
 ('Adidas', 170, 27),
 ('Nike', 170, 14),
 ('Adidas', 160, 28),
 ('Nike', 160, 31),
 ('Adidas', 150, 41),
 ('Nike', 150, 6),
 ('Adidas', 140, 36),
 ('Nike', 140, 12),
 ('Adidas', 130, 96),
 ('Nike', 130, 12),
 ('Adidas', 120, 115),
 ('Nike', 120, 16),
 ('Adidas', 110, 91),
 ('Nike', 110, 17),
 ('Adidas', 100, 72),
 ('Nike', 100, 14),
 ('Adidas', 96, 2),
 ('Nike', 95, 1),
 ('Adidas', 90, 89),
 ('Nike', 90, 13),
 ('Adidas', 86, 7),
 ('Adidas', 85, 1),
 ('Nike', 85, 5),
 ('Adidas', 80, 322),
 ('Nike', 80, 16),
 ('Nike', 79, 1),
 ('Adidas', 76, 149),
 ('Adidas', 75, 1),
 ('Nike', 75, 7),
 ('Adidas', 70, 87),
 ('Nike', 70, 4),
 ('Adidas', 66, 102),
 ('Nike', 65, 1),
 ('Adidas', 63, 1),
 ('Adidas', 60, 211),
 ('Nike', 60, 2),
 ('Adidas', 56, 174),
 ('Adidas', 55, 2),
 ('Adidas', 53, 43),
 ('Adidas', 50, 183),
 ('Nike', 50, 5),
 ('Adidas', 48, 42),
 ('Nike', 48, 1),
 ('Adidas', 46, 163),
 ('Adidas', 45, 1),
 ('Nike', 45, 3),
 ('Adidas', 43, 51),
 ('Adidas', 40, 81),
 ('Nike', 40, 1),
 ('Adidas', 38, 24),
 ('Adidas', 36, 25),
 ('Adidas', 33, 24),
 ('Adidas', 30, 37),
 ('Nike', 30, 2),
 ('Adidas', 28, 38),
 ('Adidas', 27, 18),
 ('Adidas', 25, 28),
 ('Adidas', 23, 1),
 ('Adidas', 20, 8),
 ('Adidas', 18, 4),
 ('Adidas', 16, 4),
 ('Adidas', 15, 27),
 ('Adidas', 13, 27),
 ('Adidas', 12, 1),
 ('Adidas', 10, 11),
 ('Adidas', 9, 1)]

## 3. Categorizing Price Ranges
<p>We've discovered that there are 77 unique prices for the products in our database, making the output of our last query challenging to analyze.</p>
<p>Let's enhance our previous query by assigning labels to different price ranges, grouping by <code>brand</code> and <code>label</code>. Additionally, we will include the total <code>revenue</code> for each price range and <code>brand</code>.</p>

In [ ]:
%%sql

-- Select the brand, a count of all products in the finance table, and total revenue
-- Create four labels for products based on their price range, aliasing as price_category
-- Join brands to finance on product_id and filter out products missing a value for brand
-- Group results by brand and price_category, and sort by total_revenue in descending order

SELECT
    b.brand,
    COUNT(f.*) AS product_count,
    SUM(f.revenue) AS total_revenue,
    CASE
        WHEN f.listing_price < 42 THEN 'Budget'
        WHEN f.listing_price >= 42 AND f.listing_price < 74 THEN 'Average'
        WHEN f.listing_price >= 74 AND f.listing_price < 129 THEN 'Expensive'
        ELSE 'Elite' END AS price_category
FROM finance AS f
JOIN brands AS b
    ON f.product_id = b.product_id
WHERE b.brand IS NOT NULL
GROUP BY b.brand, price_category
ORDER BY total_revenue DESC;


 * postgresql:///sports
8 rows affected.


[('Adidas', 849, 4626980.069999999, 'Expensive'),
 ('Adidas', 1060, 3233661.060000001, 'Average'),
 ('Adidas', 307, 3014316.8299999987, 'Elite'),
 ('Adidas', 359, 651661.1200000002, 'Budget'),
 ('Nike', 357, 595341.0199999992, 'Budget'),
 ('Nike', 82, 128475.59000000003, 'Elite'),
 ('Nike', 90, 71843.15000000004, 'Expensive'),
 ('Nike', 16, 6623.5, 'Average')]

## 4. Average Discount Analysis by Brand
<p>Fascinatingly, when we group products by brand and price range, it becomes evident that Adidas items generate more total revenue across all price categories! Particularly, Adidas products in the <code>"Elite"</code> category, priced at $129 or more, consistently yield the highest revenue. This suggests a potential revenue increase by adjusting the product mix to include a higher proportion of these premium items!</p>
<p>It's important to note that we have been examining <code>listing_price</code> thus far, which may not represent the final sale price. To gain a better understanding of <code>revenue</code>, let's explore the <code>discount</code>—the percentage reduction in the <code>listing_price</code> when the product is sold. This investigation aims to uncover whether there are differences in the amount of <code>discount</code> offered between brands, as it could be influencing overall <code>revenue</code>.</p>

In [ ]:
%%sql

-- Select brand and average_discount as a percentage
-- Join brands to finance on product_id
-- Aggregate by brand
-- Filter for products without missing values for brand

SELECT
    b.brand,
    AVG(f.discount) * 100 AS average_discount_percentage
FROM brands b
JOIN finance f
    ON b.product_id = f.product_id
WHERE b.brand IS NOT NULL
GROUP BY b.brand;


 * postgresql:///sports
2 rows affected.


[('Nike', 0.0), ('Adidas', 33.452427184465606)]

## 5. Examining the Correlation between Revenue and Reviews
<p>Interestingly, Nike products seem to have no offered <code>discount</code>, while Adidas products, despite generating substantial revenue, often come with significant discounts!</p>
<p>To enhance revenue further, the company might consider adjusting the discounts on Adidas products, closely monitoring sales volume for any potential impact. Alternatively, a modest discount on Nike products could be experimented with. While this might decrease the average revenue per Nike product, it could potentially boost overall revenue by driving increased sales volume.</p>
<p>Let's now investigate whether any correlations exist between the columns in our database, specifically examining the strength and direction of the correlation between <code>revenue</code> and <code>reviews</code>.</p>

In [ ]:
%%sql

-- Calculate the correlation between reviews and revenue as review_revenue_corr
-- Join the reviews and finance tables on product_id

SELECT CORR(r.reviews, f.revenue) AS review_revenue_correlation
FROM reviews r
JOIN finance f
    ON r.product_id = f.product_id;


 * postgresql:///sports
1 rows affected.


[(0.6518512283481301,)]

## 6. Exploring Ratings and Reviews Based on Product Description Length
<p>Fascinatingly, there exists a robust positive correlation between <code>revenue</code> and <code>reviews</code>. This implies that increasing the number of reviews on the company's website could potentially boost sales, particularly for items with a higher review count.</p>
<p>It's worth exploring whether the length of a product's <code>description</code> might impact its <code>rating</code> and <code>reviews</code>. If so, the company could consider establishing content guidelines for listing products on their website and conduct experiments to assess its influence on <code>revenue</code>. Let's delve into this hypothesis!</p>

In [ ]:
%%sql

-- Calculate description_length
-- Convert rating to a numeric data type and calculate average_rating
-- Join info to reviews on product_id and group the results by description_length
-- Filter for products without missing values for description, and sort results by description_length

SELECT
    TRUNC(LENGTH(i.description), -2) AS description_length,
    ROUND(AVG(CAST(r.rating AS NUMERIC)), 2) AS average_rating
FROM info i
JOIN reviews r ON i.product_id = r.product_id
WHERE i.description IS NOT NULL
GROUP BY description_length
ORDER BY description_length;


 * postgresql:///sports
7 rows affected.


[(Decimal('0'), Decimal('1.87')),
 (Decimal('100'), Decimal('3.21')),
 (Decimal('200'), Decimal('3.27')),
 (Decimal('300'), Decimal('3.29')),
 (Decimal('400'), Decimal('3.32')),
 (Decimal('500'), Decimal('3.12')),
 (Decimal('600'), Decimal('3.65'))]

## 7. Review Analysis: Monthly Trends and Brand Insights
<p>Regrettably, there doesn't seem to be a discernible pattern between the length of a product's <code>description</code> and its <code>rating</code>.</p>
<p>Given the established correlation between <code>reviews</code> and <code>revenue</code>, one strategy the company could consider is conducting experiments with various sales approaches to encourage more customer reviews on their purchases. This could involve initiatives like offering a slight discount on future purchases to incentivize reviews.</p>
<p>Now, let's delve into an analysis of the volume of <code>reviews</code> categorized by month to uncover any trends or gaps that could be strategically leveraged.</p>

In [ ]:
%%sql

-- Select brand, month from last_visited, and a count of all products in reviews aliased as num_reviews
-- Join traffic with reviews and brands on product_id
-- Group by brand and month, filtering out missing values for brand and month
-- Order the results by brand and month

SELECT
    b.brand,
    DATE_PART('month', t.last_visited) AS month,
    COUNT(r.*) AS num_reviews
FROM brands AS b
JOIN traffic AS t ON b.product_id = t.product_id
JOIN reviews AS r ON t.product_id = r.product_id
WHERE b.brand IS NOT NULL
    AND DATE_PART('month', t.last_visited) IS NOT NULL
GROUP BY b.brand, month
ORDER BY b.brand, month;


 * postgresql:///sports
24 rows affected.


[('Adidas', 1.0, 253),
 ('Adidas', 2.0, 272),
 ('Adidas', 3.0, 269),
 ('Adidas', 4.0, 180),
 ('Adidas', 5.0, 172),
 ('Adidas', 6.0, 159),
 ('Adidas', 7.0, 170),
 ('Adidas', 8.0, 189),
 ('Adidas', 9.0, 181),
 ('Adidas', 10.0, 192),
 ('Adidas', 11.0, 150),
 ('Adidas', 12.0, 190),
 ('Nike', 1.0, 52),
 ('Nike', 2.0, 52),
 ('Nike', 3.0, 55),
 ('Nike', 4.0, 42),
 ('Nike', 5.0, 41),
 ('Nike', 6.0, 43),
 ('Nike', 7.0, 37),
 ('Nike', 8.0, 29),
 ('Nike', 9.0, 28),
 ('Nike', 10.0, 47),
 ('Nike', 11.0, 38),
 ('Nike', 12.0, 35)]

## 8. Evaluation of Footwear Product Performance
<p>It appears that product reviews reach their peak in the first quarter of the calendar year, indicating an opportunity to conduct experiments and boost review volumes in the remaining nine months!</p>
<p>Thus far, our analysis has predominantly focused on Adidas vs. Nike products. Now, let's shift our focus to the nature of the products themselves. Given the absence of product type labels, we will create a Common Table Expression (CTE) to filter <code>description</code> for keywords. Subsequently, we'll utilize the results to determine the proportion of the company's inventory consisting of footwear products and assess the median <code>revenue</code> generated by these items.</p>

In [ ]:
-- Create the footwear CTE, containing description and revenue
-- Filter footwear for products with a description containing %shoe%, %trainer%, or %foot%
-- Also filter for products that are not missing values for description
-- Calculate the number of products and median revenue for footwear products

WITH footwear AS (
    SELECT
        i.description,
        f.revenue
    FROM info i
    JOIN finance f
        ON i.product_id = f.product_id
    WHERE i.description ILIKE '%shoe%'
        OR i.description ILIKE '%trainer%'
        OR i.description ILIKE '%foot%'
        AND i.description IS NOT NULL
)

SELECT
    COUNT(*) AS num_footwear_products,
    PERCENTILE_DISC(0.5) WITHIN GROUP(ORDER BY revenue) AS median_footwear_revenue
FROM footwear;


 * postgresql:///sports
1 rows affected.


[(2700, 3118.36)]

## 9. Assessment of Clothing Product Performance
<p>Recalling from the initial task, we identified 3,117 products without missing values for <code>description</code>. Among these, 2,700 are classified as footwear products, constituting approximately 85% of the company's inventory. Remarkably, footwear products boast a median revenue exceeding $3,000!</p>
<p>While this information is intriguing, we lack a benchmark to evaluate whether the <code>median_revenue</code> for footwear is considered favorable or unfavorable compared to other product categories. To address this, for our final task, let's explore how this performance contrasts with clothing products. We will leverage the existing <code>footwear</code> data, subsequently applying a filter to calculate the number of products and the <code>median_revenue</code> for items outside the <code>footwear</code> category.</p>

In [ ]:
-- Copy the footwear CTE from the previous task
-- Calculate the number of products in info and median revenue from finance
-- Inner join info with finance on product_id
-- Filter the selection for products with a description not in footwear

WITH footwear AS (
    SELECT
        i.description,
        f.revenue
    FROM info i
    JOIN finance f
        ON i.product_id = f.product_id
    WHERE i.description ILIKE '%shoe%'
        OR i.description ILIKE '%trainer%'
        OR i.description ILIKE '%foot%'
        AND i.description IS NOT NULL
)

SELECT
    COUNT(i.*) AS num_clothing_products,
    PERCENTILE_DISC(0.5) WITHIN GROUP(ORDER BY f.revenue) AS median_clothing_revenue
FROM info i
JOIN finance f
    ON i.product_id = f.product_id
WHERE i.description NOT IN (
    SELECT description
    FROM footwear);


 * postgresql:///sports
1 rows affected.


[(417, 503.82)]